crates/turbo-tasks/src/task/task_input.rs

```diff
@@ -31,7 +40,147 @@ impl TaskInput for ConcreteTaskInput {
     }
 }
 
-impl TaskInput for String {
+/// This type exists to allow swapping out the underlying string type easily.
+//
+// If you want to change the underlying string type to `Arc<str>`, please ensure that you profile
+// perforamnce. The current implementation offers very cheap `String -> RcStr -> String`, meaning we
+// only pay for the allocation for `Arc` when we pass `format!("").into()` to a function.
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
```

Comment:
Possible follow-up: Instead of using `#[derive(Default)]`, it seems like there's a good opportunity to special-case `Default` to return a globally allocated `RcStr`.